packages/apps/transfer/src/services/cross-chain-transfer-finish/get-transfer-data.ts

```diff
@@ -63,52 +59,42 @@ export async function getTransferData({
     return { error: t('Invalid length of request key') };
   }
 
-  const pactCommand = new PactCommand();
-
-  pactCommand.requestKey = requestKey;
-
   try {
     const chainInfoPromises = Array.from(new Array(20)).map((item, chainId) => {
       const host = getKadenaConstantByNetwork(network).apiHost({
         networkId: chainNetwork[network].network,
         chainId: convertIntToChainId(chainId),
       });
-      return pactCommand.poll(host);
+      const { getStatus } = getClient(host);
+      return getStatus(requestKey);
     });
     const chainInfos = await Promise.all(chainInfoPromises);
 
-    const found: { chainId: number; tx: ICommandResult } | undefined =
-      chainInfos.reduce(
-        (
-          acc: { chainId: number; tx: ICommandResult } | undefined,
-          curr: IPollResponse,
-          chain: number,
-          array: IPollResponse[],
-        ) => {
-          array.splice(chain - 1);
-          if (curr[requestKey] !== undefined) {
-            return { chainId: chain, tx: curr[validatedRequestKey] };
-          }
-        },
-        undefined,
-      );
+    const request = chainInfos.find((chainInfo) => requestKey in chainInfo);
 
-    if (found === undefined) {
+    if (!request) {
       return { error: t('No request key found') };
     }
 
+    const found = request[requestKey];
+
+    const { events, result } = found;
+
+    if ('error' in result) {
+      return result.error;
+      // return { error: ('message' in result.error ? (result.error.message as string) : 'An error occurred.' };
+    }
+
     const [senderAccount, receiverAccount, guard, targetChain, amount] =
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      found.tx?.continuation?.continuation.args as Array<any>;
+      found?.continuation?.continuation.args as Array<any>;
 
-    const { step, stepHasRollback, pactId } = found.tx
-      ?.continuation as IPactExec;
-    const { events, result } = found.tx;
+    const { step, stepHasRollback, pactId } = found?.continuation as IPactExec;
```

Comment:
All the castings (`as`) aren't great. They were here already, but would still be good to look into later.